core/tx_pool.go

```diff
@@ -425,7 +442,13 @@ func (pool *TxPool) AddBatch(txs []*types.Transaction) {
 			glog.V(logger.Debug).Infoln("tx error:", err)
 		}
 	}
-	pool.promoteExecutables()
+
+	state, err := pool.currentState()
+	if err != nil {
+		return
+	}
+
+	pool.promoteExecutables(state)
```

Comment:
Since we're ignoring possible errors here (not sure yet why), this perhaps would be a lot cleaner with:

```

if state, err := pool.currentState(); err == nil {
    pool.promoteExecutables(state)
}
```